lib/pos_blockchain.go

```diff
@@ -472,6 +488,48 @@ func (bc *Blockchain) GetUncommittedTipView() (*UtxoView, error) {
 	return utxoView, nil
 }
 
+// BuildUtxoViewToBlockParent builds a UtxoView to the block provided. It does this by
+// identifying all uncommitted ancestors of this block and then connecting those blocks.
+func (bc *Blockchain) BuildUtxoViewToBlockParent(parentHash BlockHash) (*UtxoView, error) {
```

Comment:
As an alternative name, wdyt of this? In other places, we've started to transition the naming for "UtxoView" to "BlockView". I have no strong opinion here

```suggestion
func (bc *Blockchain) buildBlockView(blockHash BlockHash) (*UtxoView, error) {
```

Reply:
Block view is a little confusing to me - it could sometimes refer to `block.Header.ProposedInView`. So for now, I think it's cleanest to still refer to these as `UtxoView` structs for now